### PR TITLE
Update date.js to handle 12PM correctly

### PR DESCRIPTION
**Pull request by MelSumner**
If merged, this PR updates the date.js file to handle 12 noon times. Otherwise it returns 00 as the hour with both AM and PM.

**Before:** 
image

**After:**
image

### CODE REVIEW

File: starter-files/date.js [hunk 1]
```diff
@@ -48,13 +48,22 @@ export function dateToString(date) {
...
-  return `${
-    MONTH_NAMES[d.getMonth()]
-  } ${d.getDate()}, ${d.getFullYear()} ${padLeadingZeroes(
-    d.getHours() % 12,
-    2
-  )}:${padLeadingZeroes(
-    d.getMinutes(),
-    2
-  )}:${padLeadingZeroes(d.getSeconds(), 2)} ${ampm}`;
+  if (d.getHours() === 12) {
```
Comment (mike-north) [inline starter-files/date.js]: I think we may need your fix for `13:00`, `14:00`, etc...

Let's combine this with the `AM`/`PM` condition. Something like

```js
let hrs = d.getHours();
let ampm = 'AM';
if (hrs > 12) {
  // afternoon
  ampm = 'PM';
  hrs = hrs % 12;
}
```

And then we don't need another condition to switch between using `d.getHours()` and `d.getHours() % 12` (`hrs` will already be what we want)

Comment (MelSumner) [reply starter-files/date.js]: I think this will leave everything in military time, but I'm not sure that's what you wanted?

**Comment by MelSumner** [comment]
So something weird- I updated the tests to check for midnight and noon: 

In date-test.js: 

```
  test('test date utility', function(assert) {
    // A QUnit Assertion
    assert.equal(
      dateToString('04/05/1983'),
      'Apr 5, 1983 00:00.00 AM',
      'MM/DD/YYYY'
    );
    assert.equal(
      dateToString('4/5/1983'),
      'Apr 5, 1983 00:00.00 AM',
      'M/D/YYYY'
    );
    assert.equal(
      dateToString('26 June 2010 13:00'),
      'Jun 26, 2010 01:00.00 PM',
      '1:00 PM'
    );
    assert.equal(
      dateToString('05 April 1983 12:00'),
      'Apr 5, 1983 12:00.00 PM',
      '12PM'
    );

    assert.equal(
      dateToString('05 April 1983 00:00'),
      'Apr 5, 1983 12:00.00 AM',
      '12AM'
    );
  });
```

This passes in Windows + Chrome but does NOT pass on Mac + Chrome: 

**Windows:** 
image

**Mac:** 
image

File: starter-files/date.js (hunk 1 above)
Comment (MelSumner) [reply starter-files/date.js]: updated the date.js util after working on it a little more.

**Pull request merged**